examples/run/cuda/seq_example_cuda.cpp

```diff
@@ -0,0 +1,282 @@
+          ----------------------------*/
+
+        // cuda
+
+        /*time*/ auto start_seeding_cuda = std::chrono::system_clock::now();
+
+        auto sa_cuda_result = sa_cuda(spacepoints_per_event);
+        auto& seeds_cuda = sa_cuda_result.second;
+        n_seeds_cuda += seeds_cuda.headers[0];
+
+        /*time*/ auto end_seeding_cuda = std::chrono::system_clock::now();
+        /*time*/ std::chrono::duration<double> time_seeding_cuda =
+            end_seeding_cuda - start_seeding_cuda;
+        /*time*/ seeding_cuda += time_seeding_cuda.count();
+
+        // cpu
+
+        /*time*/ auto start_seeding_cpu = std::chrono::system_clock::now();
+
+        traccc::host_seed_container seeds;
+        traccc::host_internal_spacepoint_container internal_sp_per_event;
+        if (!skip_cpu) {
+            auto sa_result = sa(spacepoints_per_event);
+            internal_sp_per_event = sa_result.first;
+            seeds = sa_result.second;
+            n_internal_spacepoints += internal_sp_per_event.total_size();
+        }
+        n_seeds += seeds.total_size();
+
+        /*time*/ auto end_seeding_cpu = std::chrono::system_clock::now();
+        /*time*/ std::chrono::duration<double> time_seeding_cpu =
+            end_seeding_cpu - start_seeding_cpu;
+        /*time*/ seeding_cpu += time_seeding_cpu.count();
+
+        /*----------------------------------
+          compare seeds from cpu and cuda
+          ----------------------------------*/
+
+        if (!skip_cpu) {
+            int n_match = 0;
+            for (auto seed : seeds.items[0]) {
+                if (std::find(
+                        seeds_cuda.items[0].begin(),
+                        seeds_cuda.items[0].begin() + seeds_cuda.headers[0],
+                        seed) !=
+                    seeds_cuda.items[0].begin() + seeds_cuda.headers[0]) {
+                    n_match++;
+                }
+            }
+            float matching_rate = float(n_match) / seeds.headers[0];
+            std::cout << "event " << std::to_string(skip_events + event)
+                      << " seed matching rate: " << matching_rate << std::endl;
+        }
+
+        /*------------
+             Writer
+          ------------*/
+
+        if (!skip_write && !skip_cpu) {
+
+            traccc::measurement_writer mwriter{
+                std::string("event") + event_number + "-measurements.csv"};
+            for (size_t i = 0; i < measurements_per_event.items.size(); ++i) {
+                auto measurements_per_module = measurements_per_event.items[i];
+                auto module = measurements_per_event.headers[i];
+                for (const auto& measurement : measurements_per_module) {
+                    const auto& local = measurement.local;
+                    mwriter.append({module.module, "", local[0], local[1], 0.,
+                                    0., 0., 0., 0., 0., 0., 0.});
+                }
+            }
+
+            traccc::spacepoint_writer spwriter{
```

Comment:
There are also some writer functions in [writer.hpp](https://github.com/acts-project/traccc/blob/556542b39feef4b3cda06d44f2becb6adc7fda0f/io/include/io/writer.hpp#L13).

Reply:
Didn't know that it exists! Thanks